### PR TITLE
OPENEUROPA-2660: Default thumbnails for certain media types (map, chart, social media feed)

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php [hunk 1]
```diff
@@ -40,10 +40,12 @@ class MediaSourceWebtoolsTest extends MediaSourceTestBase {
    *   The widget human name.
    * @param string $service
    *   The webtools service associated with the widget.
+   * @param string $svg_filename
+   *   The filename of svg.
```
Comment (nagyad) [inline modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php:44]: We need to rename this. Can we use something generic without the file extension?

File: modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php [hunk 2]
```diff
@@ -40,10 +40,12 @@ class MediaSourceWebtoolsTest extends MediaSourceTestBase {
...
    *
    * @dataProvider providerTestMediaWebtoolsSource
    */
-  public function testMediaWebtoolsSource(string $widget_type, string $widget_name, string $service): void {
+  public function testMediaWebtoolsSource(string $widget_type, string $widget_name, string $service, string $svg_filename): void {
```
Comment (nagyad) [inline modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php:48]: Rename the parameter.

File: modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php [hunk 3]
```diff
@@ -60,6 +62,10 @@ public function testMediaWebtoolsSource(string $widget_type, string $widget_name
     $page->pressButton('Save');
     $assert_session->addressEquals('admin/content/media');
 
+    $this->drupalGet('/media/1');
+    $img_src = $page->find('css', '.field--name-thumbnail .field__item img')->getAttribute('src');
+    $this->assertContains($svg_filename, $img_src);
```
Comment (nagyad) [inline modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php:67]: We are not using svg anymore.

File: modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php (hunk 1 above)
Comment (nagyad) [reply modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php:44]: thumbnail_filename or icon_filename, something like that.

File: modules/oe_webtools_media/oe_webtools_media.install [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * @see \media_install()
+ */
+function oe_webtools_media_install() {
+  $source = drupal_get_path('module', 'oe_webtools_media') . '/images/icons';
+  $destination = \Drupal::config('media.settings')->get('icon_base_uri');
+  /** @var \Drupal\Core\File\FileSystemInterface $file_system */
+  $file_system = \Drupal::service('file_system');
+  $file_system->prepareDirectory($destination, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS);
+
+  $files = $file_system->scanDirectory($source, '/.*\.(svg|png|jpg|jpeg|gif)$/');
```
Comment (22Alexandra) [inline modules/oe_webtools_media/oe_webtools_media.install:25]: Reading the ticket's comments the svg files cannot be styled but we still search for svg extensions. Can we keep only the extensions that we can use?

File: modules/oe_webtools_media/oe_webtools_media.post_update.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * Post update functions for OpenEuropa Webtools Media module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\File\Exception\FileException;
+use Drupal\Core\File\FileSystemInterface;
+
+/**
+ * Move png files for webtools media types into the public folder.
```
Comment (22Alexandra) [inline modules/oe_webtools_media/oe_webtools_media.post_update.php:14]: Below you're searching for more than just png files.

File: modules/oe_webtools_media/oe_webtools_media.post_update.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * @see \media_install()
+ */
+function oe_webtools_media_post_update_00001(): void {
+  $source = drupal_get_path('module', 'oe_webtools_media') . '/images/icons';
+  $destination = \Drupal::config('media.settings')->get('icon_base_uri');
+  /** @var \Drupal\Core\File\FileSystemInterface $file_system */
+  $file_system = \Drupal::service('file_system');
+  $file_system->prepareDirectory($destination, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS);
+
+  $files = $file_system->scanDirectory($source, '/.*\.(svg|png|jpg|jpeg|gif)$/');
```
Comment (22Alexandra) [inline modules/oe_webtools_media/oe_webtools_media.post_update.php:25]: As above: Can we keep only the extensions that we can use?

File: modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php [hunk 7]
```diff
@@ -132,6 +138,15 @@ protected function createWebtoolsMediaType(string $media_type_id, string $widget
     ];
     $this->createMediaTypeFields($fields, $media_type_id);
 
+    // Use the default formatter and settings for image.
+    $component = \Drupal::service('plugin.manager.field.formatter')
```
Comment (upchuk) [inline modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php:142]: Why are all these method calls on multiple lines? Can we move them to one line?

File: modules/oe_webtools_media/src/Plugin/media/Source/Webtools.php [hunk 8]
```diff
@@ -100,16 +110,33 @@ public function getWidgetTypes() {
...
     ];
   }
 
+  /**
+   * Gets the thumbnail image URI based on widget type.
+   *
+   * @return string
+   *   URI of the thumbnail or NULL if there is no specific icon.
+   */
+  protected function getThumbnail(): ?string {
```
Comment (upchuk) [inline modules/oe_webtools_media/src/Plugin/media/Source/Webtools.php:134]: Please also clean-up this method a bit. We don't need to check for the presence of the default thumbnail cause you hardcode them in the getWidgetTypes() method. So something like this:

```
  protected function getThumbnail(): ?string {
    $widget_type = $this->configuration['widget_type'];
    $icon_base = $this->configFactory->get('media.settings')->get('icon_base_uri');
    return $icon_base . '/' . $this->getWidgetTypes()[$widget_type]['default_thumbnail'];
  }
```

File: modules/oe_webtools_media/src/Plugin/media/Source/Webtools.php [hunk 9]
```diff
@@ -32,6 +33,15 @@ public function getMetadataAttributes() {
     return [];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getMetadata(MediaInterface $media, $attribute_name) {
+    if ($attribute_name === 'thumbnail_uri') {
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_media/src/Plugin/media/Source/Webtools.php:40]: yes. I have fixed my silly mistake.

**Pull request merged**